src/extension.ts

```diff
@@ -25,20 +25,30 @@ export namespace Openshift {
     }
 
     export namespace Project {
-        export const create = async function createProjectCmd(cli: cli.ICli, explorer: explorerFactory.OpenShiftExplorer) {
-            const projectName = await vscode.window.showInputBox({
+
+        export const projectNameValidation = async function projectName() {
+            const characterRegex = /[a-z0-9]([-a-z0-9]*[a-z0-9])?/;
+            const lengthofCharacterRegex = /^[0-9A-Za-z!@\.;:'"?-]{1,63}$/;
+            return await vscode.window.showInputBox({
```

Comment:
method is named `projectNameValidation` but it also request for input, that is not right. Create a method that gets a name as a parameter and returns ether `null` or error message. Then just assign that method to 'validateInput` field in input options.